llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -16752,6 +16755,22 @@ template <class MatchContextClass> SDValue DAGCombiner::visitFMA(SDNode *N) {
   return SDValue();
 }
 
+SDValue DAGCombiner::visitFMAD(SDNode *N) {
+  SDValue N0 = N->getOperand(0);
+  SDValue N1 = N->getOperand(1);
+  SDValue N2 = N->getOperand(2);
+  EVT VT = N->getValueType(0);
+  SDLoc DL(N);
+
+  // Constant fold FMAD.
+  if (isa<ConstantFPSDNode>(N0) && isa<ConstantFPSDNode>(N1) &&
```

Comment:
Should maybe use `DAG.isConstantFPBuildVectorOrConstantFP`? But I see the FMA case does not.

Reply:
I've been hoping to convert SelectionDAG::foldConstantFPMath to take a `ArrayRef<SDValue> Ops` arg instead of assuming a binop, but have never gotten around to it.